include/hffix.hpp

```diff
@@ -504,36 +505,32 @@ class message_writer {
         // Calculate and write out the BodyLength.
         // BodyLength does not include the SOH character after the BodyLength field.
         // BodyLength does not include the SOH character before the CheckSum field.
-        if (body_length_) {
-            size_t bodylength = next_ - (body_length_ + 7);
-            for(char* b = body_length_ + 5; b >= body_length_; --b) {
-                *b = '0' + (bodylength % 10);
-                bodylength /= 10;
-            }
+        if (!body_length_) {
+            throw std::logic_error("hffix message_writer.push_back_trailer called before message_writer.push_back_header");
         }
-        else throw std::logic_error("hffix message_writer.push_back_trailer called before message_writer.push_back_header");
+
+        size_t const len = next_ - (body_length_ + 7);
+        body_length_[0] = '0' + (len / 100000) % 10;
+        body_length_[1] = '0' + (len / 10000) % 10;
+        body_length_[2] = '0' + (len / 1000) % 10;
+        body_length_[3] = '0' + (len / 100) % 10;
+        body_length_[4] = '0' + (len / 10) % 10;
+        body_length_[5] = '0' + len % 10;
 
         if (buffer_end_ - next_ < 7) {
             details::throw_range_error();
         }
 
         // write out the CheckSum after optionally calculating it
         if (calculate_checksum) {
-            // This char-to-unsigned-int cast may seem odd because chars with
```

Comment:
technically this is not an overflow, it's just an *implementation defined* conversion from signed to unsigned.